control/state.py

```diff
@@ -125,9 +195,45 @@ def __init__(self, config):
             self.logger.error(f"Unable to create omap: {ex}. Exiting!")
             raise
 
+    def __exit__(self, exc_type, exc_value, traceback):
+        if self.use_notify:
+            self.watch.close()
```

Comment:
if `register_watch` was never called this will fail. Instead of checking on `use_notify` check if `watch` is `None`.